craft_cli/messages.py

```diff
@@ -551,19 +551,24 @@ def _get_progress_params(self, permanent: bool):
         return stream, use_timestamp, ephemeral
 
     @_active_guard()
-    def progress(self, text: str, permanent: bool = False) -> None:
+    def progress(self, text: str, permanent: bool = False, update_titlebar: bool = False) -> None:
```

Comment:
Thanks for the PR! I wonder if instead of using a new parameter to `Emitter.progress()` we should instead make it a new parameter in `Emitter.init()`, so that every call to `Emitter.progress()` would update the titlebar. Would that be too noisy?

Reply:
I think that it isn't a good idea because there are several "progress" messages that don't match for the title. For example:

    emit.progress(
            "Using snapcraft from snap store channel 'latest/stable' in instance "
            "because snap injection is only supported on Linux hosts.",
            permanent=True,
        )

    emit.progress(
                "Refusing to attach specified UA token as system is already attached.",
                permanent=True,
            )

    emit.progress(
                "Unable to determine CPU count; disabling parallel builds",
                permanent=True,
            )

    emit.progress(
                "--with is no longer supported, export the auth to the environment "
                f"variable {store.constants.ENVIRONMENT_STORE_CREDENTIALS!r} instead",
                permanent=True,
            )

and much more.

Reply:
you are very right!